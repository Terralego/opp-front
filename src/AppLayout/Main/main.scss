@import "../../variables.scss";

body {
  font-family: $base-font;
}

.main {
  display: flex;

  &__header {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
    max-width: $navbar-width;

    .bp3-navbar-heading {
      .logo {
        width: 50px;
      }
    }
  }
  
  &__content {
    display: flex;
    flex: auto;
    overflow: auto;
    height: 100vh;
    > div {
      width: 100%;
    }
  }
}
