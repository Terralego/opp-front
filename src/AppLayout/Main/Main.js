import React from 'react';

import Nav from '../Nav';
import Content from '../Content';

import './main.scss';

export const Main = ({ isNavBarVisible }) => (
  <main className="main">
    {isNavBarVisible && <Nav />}
    <Content />
  </main>
);

export default Main;
