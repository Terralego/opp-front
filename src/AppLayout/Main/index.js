import { connectSettingsProvider } from '../../components/SettingsProvider';

import Main from './Main';

export default connectSettingsProvider('env')(Main);
