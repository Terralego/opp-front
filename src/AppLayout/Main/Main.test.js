import React from 'react';
import renderer from 'react-test-renderer';

import Main from './Main';

jest.mock('../Nav', () => () => (<p>HeaderMock</p>));
jest.mock('../Content', () => () => (<p>ContentMock</p>));

it('should render', () => {
  const tree = renderer.create((
    <Main />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
