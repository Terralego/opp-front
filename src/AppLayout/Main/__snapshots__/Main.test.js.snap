// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<main
  className="main"
>
  <p>
    ContentMock
  </p>
</main>
`;
